store/tikv/lock_test.go

```diff
@@ -307,19 +313,79 @@ func (s *testLockSuite) TestCheckTxnStatus(c *C) {
 	// Then call getTxnStatus again and check the lock status.
 	currentTS, err = oracle.GetTimestamp(context.Background())
 	c.Assert(err, IsNil)
-	status, err = resolver.getTxnStatus(bo, txn.StartTS(), []byte("key"), currentTS, currentTS)
+	status, err = resolver.getTxnStatus(bo, txn.StartTS(), []byte("key"), currentTS, currentTS, true)
 	c.Assert(err, IsNil)
 	c.Assert(status.ttl, Equals, uint64(0))
 	c.Assert(status.commitTS, Equals, uint64(0))
 
 	// Call getTxnStatus on a committed transaction.
 	startTS, commitTS := s.putKV(c, []byte("a"), []byte("a"))
-	status, err = newLockResolver(s.store).getTxnStatus(bo, startTS, []byte("a"), currentTS, currentTS)
+	status, err = newLockResolver(s.store).getTxnStatus(bo, startTS, []byte("a"), currentTS, currentTS, true)
 	c.Assert(err, IsNil)
 	c.Assert(status.ttl, Equals, uint64(0))
 	c.Assert(status.commitTS, Equals, commitTS)
 }
 
+func (s *testLockSuite) TestCheckTxnStatusNoWait(c *C) {
+	txn, err := s.store.Begin()
+	c.Assert(err, IsNil)
+	txn.Set(kv.Key("key"), []byte("value"))
+	txn.Set(kv.Key("second"), []byte("xxx"))
+	committer, err := newTwoPhaseCommitterWithInit(txn.(*tikvTxn), 0)
+	c.Assert(err, IsNil)
+	// Increase lock TTL to make CI more stable.
+	committer.lockTTL = txnLockTTL(txn.(*tikvTxn).startTime, 200*1024*1024)
+
+	// Only prewrite the secondary key to simulate a concurrent prewrite case:
+	// prewrite secondary regions success and prewrite the primary region is pending.
+	err = committer.prewriteKeys(NewBackoffer(context.Background(), PrewriteMaxBackoff), [][]byte{[]byte("second")})
+	c.Assert(err, IsNil)
+
+	oracle := s.store.GetOracle()
+	currentTS, err := oracle.GetTimestamp(context.Background())
+	c.Assert(err, IsNil)
+	bo := NewBackoffer(context.Background(), PrewriteMaxBackoff)
+	resolver := newLockResolver(s.store)
+
+	// Call getTxnStatus for the TxnNotFound case.
+	_, err = resolver.getTxnStatus(bo, txn.StartTS(), []byte("key"), currentTS, currentTS, false)
+	c.Assert(err, NotNil)
+	_, ok := errors.Cause(err).(txnNotFoundErr)
+	c.Assert(ok, IsTrue)
+
+	errCh := make(chan error)
+	go func() {
+		errCh <- committer.prewriteKeys(bo, [][]byte{[]byte("key")})
+	}()
+
+	lock := &Lock{
+		Key:     []byte("second"),
+		Primary: []byte("key"),
+		TxnID:   txn.StartTS(),
+		TTL:     100000,
+	}
+	// Call getTxnStatusFromLock to cover the retry logic.
+	status, err := resolver.getTxnStatusFromLock(bo, lock, currentTS)
+	c.Assert(err, IsNil)
```

Comment:
At this time, the lock may still not written, we can only be sure it is written after `<-errCh`.

Reply:
The lock may still not written, and `getTxnStatusFromLock` should retry in that case, so we cover the retry logic.
The lock TTL is set to 10s, long enough for the goroutine to finish, the result should be stable.

Reply:
or we add some sleep before `errCh <- xxx` to ensure covering the retry path

Reply:
> or we add some sleep before `errCh <- xxx` to ensure covering the retry path

So as to make the CI slower?